cmd/iam-store.go

```diff
@@ -431,8 +431,41 @@ func (c *iamCache) policyDBGet(store *IAMStoreSys, name string, isGroup bool) ([
 		}
 	}
 
-	// returned policy could be empty
-	policies := mp.toSlice()
+	// returned policy could be empty, we use set to de-duplicate.
+	policies := set.CreateStringSet(mp.toSlice()...)
+
+	for _, group := range u.Credentials.Groups {
+		if store.getUsersSysType() == MinIOUsersSysType {
```

Comment:
AFAIK, at least for builtin users, Groups is always empty. In my opinion, the LoadUser() change is enough if we support builtin users.

You can add a log which prints u.Credentials.Group and run this script:

```
mc alias set myminio http://localhost:9000 minioadmin minioadmin
mc mb myminio/testbucket/
mc admin user add myminio foo foo12345
mc admin group add myminio group foo
mc admin policy attach myminio consoleAdmin --group=group
mc alias set foo http://localhost:9000 minioadmin minioadmin --api=s3v4
mc ls foo/
```